Gems/ROS2/Code/Source/RobotImporter/FixURDF/FixURDF.cpp

```diff
@@ -21,48 +21,70 @@ namespace ROS2::Utils
-            xml_node<>* inertia = urdf->document()->allocate_node(node_element, "inertia");
+            auto mass = inertial->first_node("mass");
+            if (!mass)
+            {
+                mass = urdf->document()->allocate_node(node_element, "mass");
+                mass->append_attribute(urdf->document()->allocate_attribute("value", "1."));
+                inertial->append_node(mass);
+                modified = true;
+            }
 
-            inertia->append_attribute(urdf->document()->allocate_attribute("ixx", "1."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("ixy", "0."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("ixz", "0."));
+            auto inertia = inertial->first_node("inertia");
+            if (!inertia)
+            {
+                inertia = urdf->document()->allocate_node(node_element, "inertia");
 
-            inertia->append_attribute(urdf->document()->allocate_attribute("iyx", "0."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("iyy", "1."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("iyz", "0."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("ixx", "1."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("ixy", "0."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("ixz", "0."));
 
-            inertia->append_attribute(urdf->document()->allocate_attribute("izx", "0."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("izy", "0."));
-            inertia->append_attribute(urdf->document()->allocate_attribute("izz", "1."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("iyx", "0."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("iyy", "1."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("iyz", "0."));
 
-            inertial->append_node(inertia);
+                inertia->append_attribute(urdf->document()->allocate_attribute("izx", "0."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("izy", "0."));
+                inertia->append_attribute(urdf->document()->allocate_attribute("izz", "1."));
 
-            xml_node<>* origin = urdf->document()->allocate_node(node_element, "origin");
-            origin->append_attribute(urdf->document()->allocate_attribute("xyz", "0. 0. 0."));
-            inertial->append_node(origin);
+                inertial->append_node(inertia);
+                modified = true;
+            }
 
-            auto* name = link->first_attribute("name");
-            if (name)
+            auto origin = inertial->first_node("origin");
+            if (!origin)
+            {
+                origin = urdf->document()->allocate_node(node_element, "origin");
```

Comment:
Same comment about the mass warning above

Reply:
Origin tag is [optional](http://wiki.ros.org/urdf/XML/link).  libsdformat [only uses it if available](https://github.com/gazebosim/sdformat/blob/b999425da3ea58aa659c6a9d389b8f9f6b82b9e3/src/urdf/urdf_parser/link.cpp#L281C1-L287C4).
I think there is no need for a warning, and in fact for adding this value here.